src/io/read_tree_lhalo_hdf5.c

```diff
@@ -60,15 +63,40 @@ int setup_forests_io_lht_hdf5(struct forest_info *forests_info, const int firstf
         }
         int status = fill_metadata_names(&metadata_names, run_params->TreeType);
         if (status != EXIT_SUCCESS) {
-            ABORT(status);
+            return -1;
+        }
+
+        //Read in partmass
+        if(filenr == firstfile) {
+            double partmass;
+            const double max_diff = 1e-5;
+            status = read_attribute(fd, "/Header", metadata_names.name_ParticleMass, &partmass, sizeof(partmass));
+            if(fabs(run_params->PartMass - partmass) >= max_diff) {
+                fprintf(stderr,"Error: Parameter file mentions particle mass = %g but the hdf5 file shows particle mass = %g\n",
+                        run_params->PartMass, partmass);
+                fprintf(stderr,"Diff = %g max. tolerated diff = %g\n", fabs(run_params->PartMass - partmass), max_diff);
+                fprintf(stderr,"May be the value in the parameter file needs to be updated?\n");
+                return -1;
+            }
+
+            //Check if the number of simulation output files have been set correctly
+            int32_t numsimulationfiles;
+            status = read_attribute(fd, "/Header", metadata_names.name_NumSimulationTreeFiles, &numsimulationfiles, sizeof(numsimulationfiles));
+            if(numsimulationfiles != run_params->NumSimulationTreeFiles) {
+                fprintf(stderr,"Error: Parameter file mentions total number of simulation output files = %d but the "
+                        "hdf5 field `%s' says %d tree files\n",
+                        run_params->NumSimulationTreeFiles, metadata_names.name_NumSimulationTreeFiles, numsimulationfiles);
+                fprintf(stderr,"May be the value in the parameter file needs to be updated?\n");
+                return -1;
+            }
         }
 
         int32_t nforests;
-        status = read_attribute_int(fd, "/Header", metadata_names.name_NTrees, &nforests);
+        status = read_attribute(fd, "/Header", metadata_names.name_NTrees, &nforests, sizeof(nforests));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'status' is reassigned a value before the old one has been used.](https://app.codacy.com/app/jacobseiler/sage-model/pullRequest?prid=4171404)